src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -334,6 +347,8 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
 Done:
             if (isTopLevel
                 && (emittedTypeName.ForcedArity == -1 || emittedTypeName.ForcedArity == emittedTypeName.InferredArity)
+                // Quick check so we don't have to realize the expensive `UnmangledTypeName` in the common case.
+                && emittedTypeName.TypeNameMemory.Span is ['<', ..]
```

Comment:
>'<' [](http://example.com/codeflow?start=59&length=3)

Perhaps should use a named constant with descriptive enough name.

Reply:
done.